### ISSUE TITLE
More Null Errors Latley

### ISSUE THREAD

**Issue by Zooarcher** [easysnmp/easysnmp#144]
With the pip install version, I get these..
Exception has occurred: SystemError
<built-in function set> returned NULL without setting an error

the session command reports:
error_string:'(genError) A general failure occured'

I tried installing your latest version with,
pip install git+https://github.com/kamakazikamikaze/easysnmp.git#egg=easysnmp

I get basically the same error only not reported as,
Exception has occurred: SystemError
<built-in function set> returned NULL without setting an error
but as,
'(genError) A general failure occured'

**Comment by Zooarcher** [comment]
and another error, I have to re-install the egg version to get the error, I will do shortly.

**Comment by kamakazikamikaze** [comment]
Can you provide:

* Python version
* OIDs being queried
* SNMP method(s) you're using
* Target device OS or vendor (Windows/Linux/Mac, Cisco, Arista, etc)

That would help me with replicating the setup.

**Comment by Zooarcher** [comment]
Python 3.9.2
-- not sure what you mean about OID's being queried, specifically cambium products.
Current errors are on session.set
Target device also as OID's, cambium products.. unknown os. 

I also cant seem to reproduce the other error as it's not currently getting by this one, sometimes it does.. 
but the other error actually loaded the sessions.py file ( I use vsc)

**Comment by kamakazikamikaze** [comment]
An example of an OID would be `SNMPv2-MIB::sysDescr.0` or `1.3.6.1.2.1.1.1`. It's the identifier you're trying to query or set.

**Comment by Zooarcher** [comment]
ohh, 
.1.3.6.1.4.1.161.19.3.3.18.1.3.0

**Comment by kamakazikamikaze** [comment]
Ok, thank you for supplying that. It looks like the value you're sending is [just an integer](http://www.oid-info.com/cgi-bin/display?oid=1.3.6.1.4.1.161.19.3.3.18.1.3&submit=Display&action=display).

Would it be possible for you to supply a packet capture (using wireshark or tcpdump) of the SNMP exchange so that I could locally replay the traffic and see how Easysnmp is processing the response? You can change anything sensitive (passphrase, IP, MAC) and we can privately discuss where/how to upload the file so that it's not publicly accessible.

**Comment by kamakazikamikaze** [comment]
Are you still experiencing this error? Can you provide some sample code of what you're running when you encounter this issue?

**Comment by Zooarcher** [comment]
Late Reply, been working on other things. 

I would say you can easily re-create this error by trying to access any ip with the wrong snmp string. 
Either as Read for a non-catchable systemerror, 
or when read snmp string is correct, but write snmp string is not it will return the non-catchable Null error.

**Comment by Zooarcher** [comment]
Any of the bump releases solve this issue? 
I'm still getting them quite frequently when setting params in equipment, or when monitoring a firmware upgrade and the device suddenly goes offline, for a reboot.. I have in place the mechanism to account for this, however it never reaches it due to the Null error.

**Comment by carlkidcrypto** [comment]
> Any of the bump releases solve this issue? I'm still getting them quite frequently when setting params in equipment, or when monitoring a firmware upgrade and the device suddenly goes offline, for a reboot.. I have in place the mechanism to account for this, however it never reaches it due to the Null error.

No, don't think any of the bump releases have. As @kamakazikamikaze mentioned in a previous comment. Would it be possible to get a minimal reproducible piece of code that triggers the issue? https://en.wikipedia.org/wiki/Minimal_reproducible_example
